ModelTraining/finetuning.py

```diff
@@ -0,0 +1,283 @@
+# Function to pre-process the new audio file
+def get_wave_file(wav_file):
+    '''
+    Function to load a wav file
+    '''
+    return AudioSegment.from_wav(wav_file)
+
+
+def export_wave_file(audio, begin, end, dest):
+    '''
+    Function to extract a smaller wav file based start and end duration information
+    '''
+    sub_audio = audio[begin * 1000:end * 1000]
+    sub_audio.export(dest, format="wav")
+
+
+def extract_segments(audioPath, sampleDict, destnPath, suffix):
+    '''
+    Function to exctact segments given a audio path folder and proposal segments
+    '''
+    # Listing the local audio files
+    local_audio_files = str(audioPath) + '/'
+    for wav_file in sampleDict.keys():
+        audio_file = get_wave_file(local_audio_files + wav_file)
+        for begin_time, end_time in sampleDict[wav_file]:
+            output_file_name = wav_file.lower().replace(
+                '.wav', '') + '_' + str(begin_time) + '_' + str(
+                    end_time) + suffix + '.wav'
+            output_file_path = destnPath + output_file_name
+            export_wave_file(audio_file, begin_time,
+                             end_time, output_file_path)
+def pre_process(dataPath=data_folder):
+    """
+    Function to convert new audio file containing False Negative into model-ready stream
+    Input -
+    dataPath: path to data folder
+    Output -
+    Will automatically put processed files in the 'filePath/new_samples/' folder
+    """
+
+    ## Create o/p folder 
+    local_dir = dataPath/"new_samples"
+    if os.path.exists(local_dir):
+        shutil.rmtree(local_dir)
+    os.makedirs(local_dir)
+
+    ## iterate over all wav files in new_data folder
+    four_sec_dict = {}
+    for item in (dataPath/"new_data/").glob("*.wav"):
+        max_length = get_duration(filename=item) - 1 
+        fourSecList = []
+        for i in range(int(max_length//4)):
+            fourSecList.append([i*4, (i+1)*4])
+
+        four_sec_dict[item.name] = fourSecList
+    
+    ## get model ready data
+    extract_segments(
+            str(dataPath/"new_data/"),
+            four_sec_dict,
+            str(local_dir)+'/',
+            "_Noise"
+        )
+    
+    ## Remove the original data folder
+    shutil.rmtree(dataPath/"new_data/")
+
+pre_process(Path("./data/"))
+
+"""
+Resulting Folder Structure - 
+./data/   
+    |-positive/
+        - xyz1.wav
+        - xyz2.wav
+        ...
+    |-negative/
+        - abc1.wav
+        - abc2.wav
+        ..
+    |-new_samples/
+        - asdfas.wav
+        - asdfas.wav
+        ..
+"""
+
+def data_blender(dataPath):
+    """
+    Function to blend the original data with new data
+    
+    """
+    pos_samples  = len((dataPath/'positive').ls())
+    neg_samples  = len((dataPath/'negative').ls())
+    new_neg_samples  = len((dataPath/'new_samples').ls())
+    total_neg_samples = neg_samples + new_neg_samples
+    
+    neg_img_list = pd.Series((dataPath/'negative').ls() + (dataPath/'new_samples').ls())
```

Comment:
Can we just use `neg_samples` and `new_neg_samples` from above or do we need to read again?
```suggestion
    neg_img_list = pd.Series(neg_samples + new_neg_samples)
```